tests/TESTS/common/fixtures.py

```diff
@@ -0,0 +1,187 @@
+                # Flash if a binary is provided and the board hasn't been flashed yet
+                platform = alloc.description['platform_name']
+                binary = self.binaries.get(platform)
+                if alloc.flashed is False and binary:
+                    if self.flasher is None:
+                        self.flasher = Flash()
+                    self.flasher.flash(build=binary, target_id=alloc.description["target_id"])
+                    alloc.flashed = True
+
+                # Create the serial connection
+                connection = SerialConnection(
+                    port=alloc.description["serial_port"],
+                    baudrate=self.baudrate,
+                    inter_byte_delay=self.serial_inter_byte_delay
+                )
+                connection.open()
+
+                # Create the serial device
+                serial_device = SerialDevice(connection, name)
+                serial_device.reset(duration=1)
+                serial_device.flush(1)
+
+                # Create the SerialDevice
+                alloc.ble_device = serial_device
+
+                return alloc.ble_device
+        return None
+
+    def release(self, ble_device: SerialDevice) -> None:
+        for alloc in self.allocation:
+            if alloc.ble_device == ble_device and alloc.ble_device is not None:
+                # Stop activities
+                alloc.ble_device.stop()
+                alloc.ble_device.serial.close()
+
+                # Cleanup
+                alloc.ble_device = None
+
+
+@pytest.fixture(scope="session")
+def board_allocator(
+        platforms: List[str],
+        binaries: Mapping[str, str],
+        serial_inter_byte_delay: float,
+        serial_baudrate: int,
+        command_delay: float
+):
+    yield BoardAllocator(platforms, binaries, serial_inter_byte_delay, serial_baudrate, command_delay)
+
+
+class ClientAllocator:
+
+    def __init__(self):
+        self.client = None  # type: Optional[BleakClient]
+
+    async def allocate(self, name: str) -> Optional[BleakClient]:
+        if self.client is None:
+            devices = await BleakScanner.discover()
+            for d in devices:
+                if name == d.name:
+                    self.client = BleakClient(d.address)
+                    attempts = 5
+                    while attempts > 0:
+                        try:
+                            connected = await self.client.connect()
+                            if connected is True:
+                                return self.client
+                        except BleakError:
+                            pass
+                        attempts -= 1
+        return None
+
+    async def release(self, client: BleakClient) -> None:
+        if self.client == client and self.client is not None:
+            attempts = 5
+            while attempts > 0:
+                try:
+                    disconnected = await self.client.disconnect()
```

Comment:
before you try to disconnect check if you're actually connected

Reply:
Done: 9faa00b0b7119a716e9ef7a784099ce9812278f1